inst/qview/qview.R

```diff
@@ -0,0 +1,194 @@
+#' @param .ns The namespace function that should be applied to the button's ID.
+#' @details
+#' The function returns the cancel button as raw HTML with a JavaScript statement
+#' embedded. Using raw HTML + JavaScript allows us to embed the button directly into
+#' a DataTable object.
+cancel_button_creator <- function(index, .ns) {
+  return(
+    as.character(
+      actionButton(
+        as.character(index),
+        glue::glue("Cancel"),
+        onclick = paste0('Shiny.onInputChange(\"', .ns("cancel_button"), '\",  this.id)')
+      )
+    )
+  )
+}
+
+## UI ####
+
+#' queueViewOutput - Data Table displaying the jobs currently in an R Simple Queue.
+#' @param id The Shiny ID to be associated with this object.
+#' @importFrom DT dataTableOutput
+#' @export
+queueViewOutput <- function(id) {
+  ns <- NS(id)
+  tagList(
+    DT::dataTableOutput(ns("queue_table"))
+  )
+}
+
+## Server ####
+
+#' queueView - Server-side function to call with callModule()
+#' @param input Shiny input
+#' @param output Shiny output
+#' @param session Shiny session
+#' @param queue_obj An rrsq::RSimpleQueue object that has already been initialized with $new.
+#' @param refresh_interval The frequency with which the UI should poll the queue and update, in milliseconds.
+#' @importFrom rrsq::RSimpleQueue
+#' @export
+queueView <- function(
+    input,
+    output,
+    session,
+    queue_obj = rrsq::RSimpleQueue$new(),
+    .user,
+    .refresh_interval = 3000
+  ) {
+
+  # Grab the namespace function used in the UI object.
+  # We will need this for creating the individual buttons in the list.
+  ns <- session$ns
+
+  print("Starting.")
+  print(glue::glue("User: {.user}"))
+  # browser()
+
+  # Get an RSQ object, which contains the methods needed for interacting with the Queue.
+  queue_obj = rrsq::RSimpleQueue$new()
+
+  # Initialize our data frame, "queue_data".
+  # We pull the source data with queue_obj$get_jobs, then convert it to a data frame and
+  #   sort the contents by ID (descending). We then assign this to as a component of a
+  #   reactiveValue for easy-access in other reactive contexts.
+  .rv <- reactiveValues()
+  .rv$queue_data <-
+    rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+    dplyr::filter(user == .user) %>%
+    dplyr::arrange(desc(id))
+
+  .return_values <- reactiveValues()
+
+  # Create an observe event that re-runs (via invalidation) every 1000 milliseconds.
+  # In this section, we grab an updated version of our data frame and compare it
+  #   to the one currently in use. If any differences are found, we upate the data
+  #   frame in use. This prevents the UI from constantly refreshing every time it
+  #   checks for new data.
+  observe({
+    invalidateLater(.refresh_interval)
+    new_data <-
+      rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+      dplyr::filter(user == .user) %>%
+      dplyr::arrange(desc(id))
+
+    if (
+      !isTRUE(all_equal(.rv$queue_data, new_data))
```

Comment:
is all_equal coming from rlang? don't forget to import it or namespace it if it is!

Reply:
dplyr, but good call!